### ISSUE TITLE
My configuration scenario

### ISSUE THREAD

**Issue by SanZamoyski** [emonbhuiyan/redsocks#1]
Hi!

I have OpenWRT 19-something version on tp-link router.
I needed to create transparent proxy (so everyone who connects to this router by default uses proxy and won't be blocked by local firewall).

Here is my scenario:

- configure ssh and, over, ssh:
- opkg update
- opkg install iptables iptables-mod-nat-extra redsocks ca-bundle ca-certificates libustream-openssl20150806 nano 
- service redsocks stop && mv /etc/redsocks.conf /etc/redsocks.conf.bkp && cd /etc && wget https://github.com/emonbhuiyan/Redsocks-OpenWRT/raw/main/redsocks.conf && mv /etc/init.d/redsocks /etc/init.d/redsocks.bkp && cd /etc/init.d && wget https://github.com/emonbhuiyan/Redsocks-OpenWRT/raw/main/redsocks && chmod +x /etc/init.d/redsocks
- nano /etc/redsocks.conf

```
base {
        log_debug = off;
        log_info = on;
        log = "syslog:local7";
        daemon = on;
        redirector = iptables;
}

redsocks {
        local_ip = 0.0.0.0;
        local_port = 1337;
        ip = YOUR_SOCKS5_PROXY_IP;		// <---------
        port = YOUR_SOCKS5_PROXY_PORT;		// <---------
        type = socks5;
        //login = "username";
        //password = "password";
}
```

- service redsocks enable
- service redsocks start
- service redsocks reload

- in http://192.168.1.1/cgi-bin/luci/admin/system/startup (Local startup) add before 'exit 0'

> service redsocks restart
> iptables -t nat -A PREROUTING -s 192.168.0.0/24 -i br-lan -p tcp --syn -j ACCEPT
> iptables -t nat -A PREROUTING -s 192.168.1.0/24 -i br-lan -p tcp --syn -j ACCEPT
> iptables -t nat -A PREROUTING -s  YOUR_SOCKS5_PROXY_IP_WITH_0_ON_END/24 -i br-lan -p tcp --syn -j ACCEPT
> iptables -t nat -A PREROUTING -i br-lan -p tcp --syn -j REDIRECT --to-ports 1337

Maybe someone will make use of it.

**Comment by animer3009** [comment]
@SanZamoyski what about UDP traffic?

**Comment by SanZamoyski** [comment]
Well, thats good idea to add. I'll test it some day...

**Comment by emonbhuiyan** [comment]
Hello @SanZamoyski ,
Just now I noticed your issue. Thanks for letting me know. I will try to test it.

**Comment by exxojay** [comment]
@SanZamoyski 
how can i contact you regarding this thread thanks

**Comment by emonbhuiyan** [comment]
Hi @SanZamoyski,

Thank you for sharing your manual configuration scenario!

We have recently updated the project to include a full **LuCI Web UI** configuration interface.

All of these configuration steps (setting up interface redirections, local ports, SOCKS5 servers, and private IP bypasses) are now fully automated using OpenWrt's native UCI configuration system and modern `procd` init scripts. You can manage everything directly from your router's browser interface under **Services -> Redsocks Proxy**.

I will close this issue now as it is fully resolved and automated by this new update. Feel free to open a new issue if you have any feedback on the Web UI!
